shotover-proxy/src/server.rs

```diff
@@ -401,7 +414,11 @@ impl<C: Codec + 'static> Handler<C> {
     ///
     /// When the shutdown signal is received, the connection is processed until
     /// it reaches a safe state, at which point it is terminated.
-    pub async fn run(&mut self, stream: TcpStream) -> Result<()> {
+    pub async fn run(
+        &mut self,
+        stream: TcpStream,
+        limiter: Arc<Option<RateLimiter<NotKeyed, InMemoryState, DefaultClock, NoOpMiddleware>>>,
```

Comment:
We should use an Option<Arc<T>> instead of Option<Arc<T>>.
That way it doesnt need to allocate when None.